### PR TITLE
v7 BrowserStack add android browsers

### PR DESCRIPTION
**Pull request by Sakchai-Refinitiv**
## Description

- Add Android to supported browsers.
- Fix the edge case width and height on mobile devices which makes test cases fail.

## Type of change

- [x] Refactor (improves code without changing its functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by bualoy-napat — APPROVED** [review]
Look good to me

File: .github/actions/run-test/action.yml [hunk 1]
```diff
@@ -41,7 +40,7 @@ runs:
       # Prevent NX caching for mode `all` because it must run all testing, even though the code has not changed.
       run: |
         [[ "${{ inputs.mode }}" == "all" ]] && export NX_SKIP_NX_CACHE=true && unset NX_BASE
-        npm run test:${{ inputs.mode }} -- --browserstack=${{ inputs.browsers }} --includeCoverage=${{ inputs.coverage }} --output=minimal --parallel=${{ inputs.parallel }}
+        npm run test:${{ inputs.mode }} -- --browserstack ${{ inputs.browsers }} --includeCoverage=${{ inputs.coverage }} --output=minimal --parallel=${{ inputs.parallel }}
```
Comment (bualoy-napat) [inline .github/actions/run-test/action.yml:43]: why we need to remove `=` ?

File: scripts/tests/run.mjs [hunk 2]
```diff
@@ -89,17 +93,44 @@ if (useBrowserStack) {
...
+    'firefox': 'Firefox',
+    'safari': 'Webkit'
+  };
+
+  /**
+   * Check the launcher is test on latest desktop browser
+   * @param {Object} BrowserStack launcher config
+   * @returns boolean
+   */
+  const isLatestDestopBrowser = (launcher) => {
```
Comment (bualoy-napat) [inline scripts/tests/run.mjs]:
```suggestion
  const isLatestDesktopBrowser = (launcher) => {
```

File: scripts/tests/run.mjs [hunk 3]
```diff
@@ -89,17 +93,44 @@ if (useBrowserStack) {
...
     // Create browserName to show as a label in the progress bar reporter
     let browserName = `${launcher.browser ?? launcher.browserName ?? launcher.device ?? 'unknown'}${launcher.browser_version ? ` ${launcher.browser_version}` : ''}` + ` (${launcher.os} ${launcher.os_version})`;
     browserName = browserName.charAt(0).toUpperCase() + browserName.slice(1);
-
-    // Safari has the connection issue and test cases failed with BrowserStack, need to test Safari on PlayWright for now.
-    if (launcher.browser.startsWith('safari')) {
-      browsers.push(playwrightLauncher({ product: 'webkit' }, { headless: true }));
+    // Default desktop browsers (latest) must use Playwright launcher in the default config
+    let browserLauncher = undefined;
+    if (isLatestDestopBrowser(launcher)) {
```
Comment (bualoy-napat) [inline scripts/tests/run.mjs]:
```suggestion
    if (isLatestDesktopBrowser(launcher)) {
```

**Review by bualoy-napat — APPROVED** [review]
